source/common/http/header_map_impl.h

```diff
@@ -99,6 +110,16 @@ class HeaderMapImpl : public HeaderMap, NonCopyable {
   void dumpState(std::ostream& os, int indent_level = 0) const override;
 
 protected:
+  struct HeaderEntryImpl;
+  using HeaderNode = std::list<HeaderEntryImpl>::iterator;
+#if HEADER_MAP_USE_FLAT_HASH_MAP
+  using HeaderNodeVector = std::vector<HeaderNode>;
```

Comment:
I would try https://github.com/abseil/abseil-cpp/blob/master/absl/container/inlined_vector.h here with a default size of 1 which would be the common case?

Reply:
good idea; I'll give that a try.